src/server/scripts/Battlegrounds/TempleOfKotmogu/battleground_temple_of_kotmogu.cpp

```diff
@@ -0,0 +1,630 @@
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AreaTrigger.h"
+#include "AreaTriggerAI.h"
+#include "Battleground.h"
+#include "BattlegroundScript.h"
+#include "Creature.h"
+#include "CreatureAI.h"
+#include "GameObject.h"
+#include "Map.h"
+#include "ObjectAccessor.h"
+#include "ObjectGuid.h"
+#include "Player.h"
+#include "ScriptMgr.h"
+#include "SpellAuraEffects.h"
+#include "SpellAuras.h"
+#include "SpellScript.h"
+#include "TaskScheduler.h"
+#include "WorldStateMgr.h"
+
+namespace TempleOfKotmogu
+{
+    namespace Achievements
+    {
+        static constexpr uint32 Powerball = 6950;
+    }
+
+    namespace Actions
+    {
+        static constexpr uint32 CheckScore = 1;
+        static constexpr uint32 ConsumeAreaTrigger = 2;
+    }
+
+    namespace AreaTriggerCreateProperties
+    {
+        static constexpr uint32 PvpRuneRejuv = 200000000;
+    }
+
+    namespace BroadcastTexts
+    {
+        static constexpr uint32 BlueOrbTaken = 62279;
+        static constexpr uint32 OrangeOrbTaken = 62280;
+        static constexpr uint32 GreenOrbTaken = 62281;
+        static constexpr uint32 PurpleOrbTaken = 62282;
+    }
+
+    namespace Creatures
+    {
+        static constexpr uint32 Trigger = 58951; // used for sounds, texts, etc.
+    }
+
+    namespace CreatureTexts::Trigger
+    {
+        static constexpr uint8 OrangeOrbReturned = 0;
+        static constexpr uint8 BlueOrbReturned = 1;
+        static constexpr uint8 PurpleOrbReturned = 2;
+        static constexpr uint8 GreenOrbReturned = 3;
+    }
+
+    namespace GameObjects
+    {
+        static constexpr uint32 GreatWall = 215760;
+        static constexpr uint32 Collision = 215787;
+
+        static constexpr uint32 OrbOfPowerBlue = 212091;
+        static constexpr uint32 OrbOfPowerPurple = 212092;
+        static constexpr uint32 OrbOfPowerGreen = 212093;
+        static constexpr uint32 OrbOfPowerOrange = 212094;
+    }
+
+    namespace Positions
+    {
+        static constexpr Position HealingBuff1 = { 1868.2396f, 1391.1198f, 11.52903f, 6.2601f };
+        static constexpr Position HealingBuff2 = { 1697.9149f, 1275.7673f, 12.16028f, 3.1154f };
+        static constexpr Position HealingBuff3 = { 1698.4271f, 1391.7951f, 11.75756f, 3.1675f };
+        static constexpr Position HealingBuff4 = { 1867.8836f, 1274.8872f, 11.85079f, 0.0331f };
+
+        static constexpr std::array<Position, 4> HealingBuffs
```

Comment:
Is there any reason why you haven't include the positions above directly in the array?